doc/classes/CanvasItem.xml

```diff
@@ -572,6 +572,19 @@
 		</member>
 		<member name="modulate" type="Color" setter="set_modulate" getter="get_modulate" default="Color(1, 1, 1, 1)">
 			The color applied to this [CanvasItem]. This property does affect child [CanvasItem]s, unlike [member self_modulate] which only affects the node itself.
+			[b]Note:[/b] It will have no effect if [member material] is set with a [ShaderMaterial], and its [member Shader.code] modifies [code]COLOR[/code] in [code]fragment()[/code]. To retify this, use a varying to store the [code]COLOR[/code] value in [code]vertex()[/code] first.
```

Comment:
```suggestion
			[b]Note:[/b] Shader materials assigned to [member material] can override this value for the final render. You can preserve the original color by storing it during the `vertex()` pass.
```

This should simplify the text a bit, and also make sure it doesn't exclude visual shaders. I'm not an expert if the advice given here is actually good though.

Reply:
Should be `[code]vertex()[/code]` still no?

Reply:
Yes, of course, sorry!